tests/js-custom-authorization/custom_authorization.py

```diff
@@ -914,6 +921,31 @@ def was_cached(response):
     return network
 
 
+def test_caching_of_app_code(network, args):
+    primary, _ = network.find_nodes()
+    with primary.client() as c:
+        LOG.info(
+            "Testing that interpreter reuse does not persist functions past app update"
+        )
+
+        def set_app_with_placeholder(new_val):
+            LOG.info(f"Replacing placeholder with {new_val}")
+            bundle = network.consortium.read_bundle_from_dir(args.js_app_bundle)
+            # Replace placeholder blindly, in the raw bundle JSON as string
+            s = json.dumps(bundle).replace("<func_caching_placeholder>", new_val)
+            bundle = json.loads(s)
+            network.consortium.set_js_app_from_bundle(primary, bundle)
+
+        for _ in range(5):
+            v = str(uuid.uuid4())
+            set_app_with_placeholder(v)
+            r = c.get("/app/func_caching")
```

Comment:
That only tests the primary, it would be good to check that the backups also receive the update when we expect (wait for commit level to reach the txid of the replacement before checking obvs), if we want to be really thorough here.